server/controller/index.js

```diff
@@ -57,3 +57,54 @@ exports.getId = async (ctx) => {
       return ctx.body;
     });
 };
+
+exports.vote = async (ctx) => {
+  const { user_id } = ctx.request.body;
+
+  if (!user_id) {
+    ctx.body = { error: 'user_id is required' };
+    return ctx.body;
+  }
+
+  await Question.findOne({ _id: ctx.params.id })
+    .then(async (res) => {
+      if (!res) {
+        ctx.body = { error: 'Question ID not found' };
+        return ctx.body;
+      }
+
+      const userVoted = res.voted_by.includes(user_id);
+
+      const updates = userVoted
+        ? {
+          $pull: { voted_by: user_id },
+          $inc: { votes: -1 },
+        }
+        : {
+          $addToSet: { voted_by: user_id },
+          $inc: { votes: 1 },
+        };
+
+      await Question.findOneAndUpdate(
+        { _id: ctx.params.id },
+        updates,
+        { new: true },
+      )
+        .then((question) => {
+          ctx.body = question;
```

Comment:
At the moment we are returning an array of `user_id`'s is it possible that we instead have mongoose just return a count of the length of the array in place of that array? I think that voting should be anonymous at least to other users.